replication.go

```diff
@@ -88,3 +88,14 @@ func (r *UpdateReplicationRequest) OK() error {
 
 	return nil
 }
+
+// ReplicationHTTPConfig contains all info needed by a client to make HTTP requests against the
+// remote bucket targeted by a replication.
+type ReplicationHTTPConfig struct {
```

Comment:
Moved this struct definition out of `replications/internal` since it now needs to be imported and used in a couple of different packages within `replications` to prevent a circular dependency.